Formula/john-jumbo.rb

```diff
@@ -1,11 +1,52 @@
 class JohnJumbo < Formula
   desc "Enhanced version of john, a UNIX password cracker"
   homepage "https://www.openwall.com/john/"
-  url "https://openwall.com/john/k/john-1.9.0-jumbo-1.tar.xz"
-  version "1.9.0"
-  sha256 "f5d123f82983c53d8cc598e174394b074be7a77756f5fb5ed8515918c81e7f3b"
   license "GPL-2.0-or-later"
   revision 1
+  stable do
+    url "https://openwall.com/john/k/john-1.9.0-jumbo-1.tar.xz"
+    version "1.9.0"
+    sha256 "f5d123f82983c53d8cc598e174394b074be7a77756f5fb5ed8515918c81e7f3b"
+    # Fixed setup `-mno-sse4.1` for some machines.
+    # See details for example from here: https://github.com/openwall/john/pull/4100
+    patch do
+      url "https://github.com/openwall/john/commit/a537bbca37c1c2452ffcfccea6d2366447ec05c2.patch?full_index=1"
+      sha256 "bb6cfff297f1223dd1177a515657b8f1f780c55f790e5b6e6518bb2cb0986b7b"
+    end
+
+    # Fixed setup of openssl@1.1 over series of patches
+    # See details for example from here: https://github.com/openwall/john/pull/4101
+    patch do
+      url "https://github.com/openwall/john/commit/4844c79bf43dbdbb6ae3717001173355b3de5517.patch?full_index=1"
+      sha256 "8469b8eb1d880365121491d45421d132b634983fdcaf4028df8ae8b9085c98ae"
+    end
+    patch do
+      url "https://github.com/openwall/john/commit/26750d4cff0e650f836974dc3c9c4d446f3f8d0e.patch?full_index=1"
+      sha256 "43d259266b6b986a0a3daff484cfb90214ca7f57cd4703175e3ff95d48ddd3e2"
+    end
+    patch do
+      url "https://github.com/openwall/john/commit/f03412b789d905b1a8d50f5f4b76d158b01c81c1.patch?full_index=1"
+      sha256 "65a4aacc22f82004e102607c03149395e81c7b6104715e5b90b4bbc016e5e0f7"
+    end
+
+    # Upstream M1/ARM64 Support.
+    # Combined diff of the following four commits, minus the doc changes
+    # that block this formula from using these commits otherwise.
+    # https://github.com/openwall/john/commit/d6c87924b85323b82994ce01724d6e458223fd36
+    # https://github.com/openwall/john/commit/d531f97180a6e5ae52e21db177727a17a76bd2b4
+    # https://github.com/openwall/john/commit/c9825e688d1fb9fdd8942ceb0a6b4457b0f9f9b4
+    # https://github.com/openwall/john/commit/716279addd5a0870620fac8a6e944916b2228cc2
+    patch do
+      url "https://raw.githubusercontent.com/Homebrew/formula-patches/50a00afbf4549fbc0ffd3855c884f7d045cf4f93/john-jumbo/john_jumbo_m1.diff"
+      sha256 "6658f02056fd6d54231d3fdbf84135b32d47c09345fc07c6f861a1feebd00902"
+    end
+
+    # Upstream GCC-11 patch due to alignment issues.
+    patch do
+      url "https://github.com/openwall/john/commit/8152ac071bce1ebc98fac6bed962e90e9b92d8cf.patch?full_index=1"
+      sha256 "efb4e3597c47930d63f51efbf18c409f436ea6bd0012a4290b05135a54d7edd4"
+    end
```

Comment:
I think this should be linux only right?

Reply:
Interesting one.  I think it should be left in since it fixes a code issue (no dependency on platform in the upstream version of code) and doesn't impact Mac builds.  If I were to build john-jumbo (disabling OpenCL in configure step due to it picking up OpenCL framework) and specifying `--cc=gcc-12` I can see the same error as reported so it *can* impact Mac so not just a Linux issue.  If reviewers feel it should only be Linux I can take another look.